core/src/impl/Kokkos_Core.cpp

```diff
@@ -150,6 +150,24 @@ int get_device_count() {
 #endif
 }
 
+void device_mem_info(size_t* free, size_t* total) {
+#if defined(KOKKOS_ENABLE_CUDA)
+  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaMemGetInfo(free, total));
+#elif defined(KOKKOS_ENABLE_HIP)
+  KOKKOS_IMPL_HIP_SAFE_CALL(hipMemGetInfo(free, total));
+#elif defined(KOKKOS_ENABLE_SYCL)
+  std::vector<sycl::device> devices = Kokkos::Impl::get_sycl_devices();
+  for (auto& dev : devices) {
```

Comment:
Why would sum over all devices? That's not what you are doing for `Cuda` or `HIP`. Also, you are not setting the values to zero before adding to it.
I'd prefer for the function to take an execution space instance (since `Kokkos` doesn't have a device abstraction) and return the free and total memory, say using a `std::pair` or so.